src/main/java/com/google/sps/data/TSP.java

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.*;
+
+/** Represents a comment in store or user page. */
+public class TSP {
+
+  // private Map<String, Integer> indexToStoreId = new HashMap<>();
+
+  public static double[][] constructAdjacencyMatrix(
+      List<Map<String, String>> coffeeShops, double userLat, double userLng) {
+    // There are size of coffee shops + 1 (user location) points in total
+    double[][] adjacencyMatrix = new double[coffeeShops.size() + 1][coffeeShops.size() + 1];
+
+    for (int i = 0; i < coffeeShops.size(); i++) {
+      adjacencyMatrix[0][i + 1] = Double.parseDouble(coffeeShops.get(i).get("distance"));
+    }
+    for (int i = 0; i < coffeeShops.size(); i++) {
+      for (int j = 0; j < coffeeShops.size(); j++) {
+        double lat1 = Double.parseDouble(coffeeShops.get(i).get("lat"));
+        double lon1 = Double.parseDouble(coffeeShops.get(i).get("lng"));
+        double lat2 = Double.parseDouble(coffeeShops.get(j).get("lat"));
+        double lon2 = Double.parseDouble(coffeeShops.get(j).get("lng"));
+        adjacencyMatrix[i + 1][j + 1] = haversineDistance(lat1, lon1, lat2, lon2);
+      }
+    }
+    return adjacencyMatrix;
+  }
+
+  /**
+   * /* Calculates the straight line distance between two sets of latitutes and longitutes A
```

Comment:
1) nit: fix typos: latitudes, longitudes
2) nit: "A breakdown of ..." should be a new sentence, preceded by a period.